Adapter/Adapter_Configuration.php

```diff
@@ -40,4 +41,40 @@ public function get($key)
             return Configuration::get($key);
         }
     }
+
+    /**
+     * Set a data in the persistence layer (Cookie for now...)
+     *
+     * @param string $key
+     * @param mixed $value
+     * @throws Core_Foundation_Exception_Exception If the size exceeds 3kB weight after json_encode.
+     * @return Adapter_Configuration $this, for fluent chaining setter.
+     */
+    public function persistUserData($key, $value)
```

Comment:
Very cool idea!
But since Configuration is meant for global shop settings that do not change often, perhaps this should be moved so something called, say, `SessionStorage`.